File: pytools/examples/cpp/client/protocol/ReqChatGm.cpp
#ifndef _REQ_CHAT_GM_
#define _REQ_CHAT_GM_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class ReqChatGm
{
private:
	String m_content;


public:
	Packet Encode()
	{
		Packet packet;
		packet.WriteString(m_content);
		packet.Encode(Msg::P_REQ_CHAT_GM);
		return packet;
	}


	void SetContent(String content)
	{
		m_content = content;
	}

};

#endif

File: pytools/examples/java/client/protocol/ReqChatGm.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqChatGm
{
	private String content;


	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeString(content);
		return packet.encode((short)Msg.P_REQ_CHAT_GM);
	}



	public void setContent(String content)
	{
		this.content = content;
	}

}
